Mage.Sets/src/mage/cards/a/AugmenterPugilist.java

```diff
@@ -0,0 +1,106 @@
+
+import java.util.UUID;
+
+/**
+ *
+ * @author htrajan
+ */
+public final class AugmenterPugilist extends ModalDoubleFacesCard {
+
+    public AugmenterPugilist(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo,
+                new CardType[]{CardType.CREATURE}, new SubType[]{SubType.TROLL, SubType.DRUID}, "{1}{G}{G}",
+                "Echoing Equation", new CardType[]{CardType.SORCERY}, new SubType[]{}, "{3}{U}{U}");
+        
+        // 1.
+        // Augmenter Pugilist
+        // Creature — Troll Druid
+        this.getLeftHalfCard().setPT(3, 3);
+
+        // Trample
+        this.getLeftHalfCard().addAbility(TrampleAbility.getInstance());
+
+        // As long as you control eight or more lands, Augmenter Pugilist gets +5/+5.
+        this.getLeftHalfCard().addAbility(new SimpleStaticAbility(
+            Zone.BATTLEFIELD,
+            new ConditionalContinuousEffect(
+                    new BoostSourceEffect(
+                            5, 5, Duration.WhileOnBattlefield
+                    ),
+                    new PermanentsOnTheBattlefieldCondition(
+                            StaticFilters.FILTER_CONTROLLED_PERMANENT_LAND,
+                            ComparisonType.MORE_THAN, 7
+                    ),
+                    "as long as you control eight or more lands, {this} gets +5/+5"
+            )
+        ));
+
+        // 2.
+        // Echoing Equation
+        // Sorcery
+
+        // Choose target creature you control. Each other creature you control becomes a copy of it until end of turn, except those creatures aren’t legendary if the chosen creature is legendary.
+        this.getRightHalfCard().getSpellAbility().addEffect(new EchoingEquationEffect());
+        this.getRightHalfCard().getSpellAbility().addTarget(new TargetControlledCreaturePermanent());
+
+    }
+
+    private AugmenterPugilist(final AugmenterPugilist card) {
+        super(card);
+    }
+
+    @Override
+    public AugmenterPugilist copy() {
+        return new AugmenterPugilist(this);
+    }
+}
+
+class EchoingEquationEffect extends OneShotEffect {
+
+    public EchoingEquationEffect() {
+        super(Outcome.Benefit);
+        staticText = "choose target creature you control. Each other creature you control becomes a copy of it until end of turn, except those creatures aren't legendary if the chosen creature is legendary";
+    }
+
+    EchoingEquationEffect(EchoingEquationEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public EchoingEquationEffect copy() {
+        return new EchoingEquationEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Permanent copyFrom = game.getPermanent(source.getFirstTarget());
+        game.getBattlefield().getAllPermanents().stream()
+            .filter(permanent -> permanent.isCreature() && !permanent.getId().equals(copyFrom.getId()) && permanent.isControlledBy(source.getControllerId()))
```

Comment:
Never use `getAllPermanents` (it's ignore a range and a phasing). Use `getAllActivePermanents(controller)` in your use case.

Reply:
You should probably just use 
```
game.getBattlefield().getActivePermanents(StaticFilters.FILTER_CONTROLLED_CREATURE, source.getControllerId(), source.getSourceId(), game)
```
since then you only have to check that it's not the source permanent

Reply:
fixed -- used `getAllActivePermanents(source.getControllerId())`